src/Api/Controllers/CollectionsController.cs

```diff
@@ -272,6 +308,33 @@ private async Task<Collection> GetCollectionAsync(Guid id, Guid orgId)
         return collection;
     }
 
+    private bool FlexibleCollectionsIsEnabled()
+    {
+        return _featureService.IsEnabled(FeatureFlagKeys.FlexibleCollections, _currentContext);
+    }
+
+    private void DeprecatedPermissionsGuard()
+    {
+        if (FlexibleCollectionsIsEnabled())
+        {
+            throw new FeatureUnavailableException("Flexible Collections is ON when it should be OFF.");
+        }
+    }
+
+    [Obsolete("Pre-Flexible Collections logic. Will be replaced by CollectionsAuthorizationHandler.")]
```

Comment:
The rest of the code in this file is largely the existing logic restored from `master`, although with the `DeprecatedPermissionsGuard()` to ensure that it's not called accidentally. I also moved `DeleteAnyCollection` here rather than putting it back in `CurrentContext` where others might call it.